api/client.go

```diff
@@ -283,22 +298,25 @@ func handleResponse(resp *http.Response, data interface{}) error {
 }
 
 func handleHTTPError(resp *http.Response) error {
-	var message string
-	var parsedBody struct {
-		Message string `json:"message"`
+	httpError := HTTPError{
+		StatusCode: resp.StatusCode,
+		RequestURL: resp.Request.URL,
 	}
+
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return err
+		httpError.Message = err.Error()
+		return httpError
 	}
-	err = json.Unmarshal(body, &parsedBody)
-	if err != nil {
-		message = string(body)
-	} else {
-		message = parsedBody.Message
+
+	var parsedBody struct {
+		Message string `json:"message"`
+	}
+	if err := json.Unmarshal(body, &parsedBody); err == nil {
+		httpError.Message = parsedBody.Message
```

Comment:
You no longer pass `body` as error `Message` in case `json.Unmarshal` is unsuccessful. Is this intentional?

Reply:
Yes, this is intentional. I remembered that the body of some 5xx responses might be a HTML document (caused by e.g. an outage) and we wouldn't want to show that as error message. For non-JSON responses, I'm fine if we just report `HTTP {StatusCode}` on stderr.